docs/src/reference/message-correlation.md

```diff
@@ -1,35 +1,43 @@
 # Message Correlation
 
-Message correlation describes how a message is correlated to a workflow instance (i.e. to a message catch event).
+You can start workflow instances with a message, by placing a message start event in the process diagram. In the case of a message start event, correlation is simple. Incoming messages are correlated with all workflows whose message start event message name matches the name of the incoming message, and instances of those workflows are started.
 
-In Zeebe, a message is not sent to a workflow instance directly. Instead, it is published with correlation information. When a workflow instance is waiting at a message catch event which specifies the same correlation information then the message is correlated to the workflow instance.
+Incoming messages can also be correlated with specific already-running workflow instances. You can target running workflow instances with messages that provide more information and can advance the process flow (message catch events), interrupt the process flow (interrupting boundary message events), or spawn new sub-tasks in the process flow (non-interrupting boundary message events).
 
-The correlation information contains the name of the message and the correlation key.
+In Zeebe, a message is not sent to a workflow instance directly. Instead, it is buffered by the broker for the duration of its TTL and correlated with any workflows where there is a match. In addition to the message name, the message correlation key can be used by the message publisher to target specific running workflow instances. The broker uses the message name to correlate the message with message events in workflows, and the value of the message correlation key to target specific running instances.
+
+Message catch and boundary events in process definitions are linked to message specifications that have a `name` (the message name) and a `correlationKey`. In the case of the message specification, the `correlationKey` is the _name of the variable_ to match against. The correlationKey in the message specification is a pointer to a variable in the workflow state.
```

Comment:
This line confuses me a bit, but I'm not sure how I'd change it. Maybe it's because the configuration step in the process definition isn't mentioned?